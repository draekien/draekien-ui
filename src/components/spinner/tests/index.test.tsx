/** @jsx jsx */
import { render } from '@testing-library/react';
import { jsx } from 'theme-ui';
import Spinner from '..';

describe('Spinner component', () => {
  test('should match snapshot when default', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when bars', () => {
    const { container } = render(<Spinner variant="bars" size="large" />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when dots', () => {
    const { container } = render(<Spinner variant="dots" size="large" />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot with custom prop values', () => {
    const { container } = render(
      <Spinner size="large" color="secondary" fullSize />
    );

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when full size and not circle', () => {
    const { container } = render(<Spinner variant="bars" fullSize />);

    expect(container).toMatchSnapshot();
  });
});
