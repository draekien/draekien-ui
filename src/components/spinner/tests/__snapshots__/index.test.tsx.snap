// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should match snapshot when bars 1`] = `
<div>
  <div
    class="css-hrragb-Spinner"
  >
    <div
      class="css-5vuwcy"
    />
    <div
      class="css-5vuwcy"
    />
    <div
      class="css-5vuwcy"
    />
  </div>
</div>
`;

exports[`Spinner component should match snapshot when default 1`] = `
<div>
  <svg
    class="css-cdso7f-Spinner"
    fill="none"
    role="img"
    stroke="currentColor"
    viewBox="0 0 32 32"
  >
    <title>
      Loading...
    </title>
    <circle
      cx="16"
      cy="16"
      opacity="0.125"
      r="12"
      stroke-width="4"
    />
    <circle
      class="css-1c39bx9-Spinner"
      cx="16"
      cy="16"
      r="12"
      stroke-dasharray="75.39822368615503"
      stroke-dashoffset="56.548667764616276"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Spinner component should match snapshot when dots 1`] = `
<div>
  <div
    class="css-hrragb-Spinner"
  >
    <div
      class="css-ug31l5"
    />
    <div
      class="css-ug31l5"
    />
    <div
      class="css-ug31l5"
    />
  </div>
</div>
`;

exports[`Spinner component should match snapshot when full size and not circle 1`] = `
<div>
  <div
    class="css-1v4x0y5-Spinner"
  >
    <div
      class="css-hrragb-Spinner"
    >
      <div
        class="css-1ilw9uo"
      />
      <div
        class="css-1ilw9uo"
      />
      <div
        class="css-1ilw9uo"
      />
    </div>
  </div>
</div>
`;

exports[`Spinner component should match snapshot with custom prop values 1`] = `
<div>
  <div
    class="css-1v4x0y5-Spinner"
  >
    <svg
      class="css-1bnas66-Spinner"
      fill="none"
      role="img"
      stroke="currentColor"
      viewBox="0 0 32 32"
    >
      <title>
        Loading...
      </title>
      <circle
        cx="16"
        cy="16"
        opacity="0.125"
        r="12"
        stroke-width="4"
      />
      <circle
        class="css-1c39bx9-Spinner"
        cx="16"
        cy="16"
        r="12"
        stroke-dasharray="75.39822368615503"
        stroke-dashoffset="56.548667764616276"
        stroke-width="4"
      />
    </svg>
  </div>
</div>
`;
